### DISCUSSION TITLE
issue: FieldPath not working in 8.0.0-alpha.4

### DISCUSSION THREAD

**Discussion by carlosbaraza** [react-hook-form/react-hook-form#8234] [Q&A]
### Version Number

8.0.0-alpha.4

### Codesandbox/Expo snack

https://codesandbox.io/s/practical-silence-ztl9wf?file=/src/App.tsx:271-484

### Steps to reproduce

1. Open codesanbox and see the failing type

### Expected behaviour

- The FieldPath API to continue to work?
- Is the user expected to use Lazy.FieldPath?

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
Considering v8 is not live, let's discuss it here. 

`const path: Auto.FieldPath<FormValues, string> = "firstName";`

You can use the above which is currently not documented anywhere considering we are still in alpha, the reason why we are changing from the current state none lazy FieldPath is considered for large and complex projects/forms, where form level can be "quite deeply nested", so if you can think of V7 FieldPath will have to generate an entire field path of every possible field path combination.  With v8, field path will be lazily evaluated.

**Reply by carlosbaraza** [reply]
Would it make sense to replace the default `FieldPath` with the new `Auto.FieldPath` and add a default `string` to the generic so the previous API keeps working with lazy evaluation?

**Reply by badsyntax** [reply]
@bluebill1049 is `Auto.FieldPath` supposed to support circular references? for example: https://codesandbox.io/s/condescending-http-gbe24i?file=/src/test.ts
